src/algorithms/pid/IrtCherenkovParticleID.cc

```diff
@@ -0,0 +1,398 @@
+// Copyright 2023, Christopher Dilks, adapted from Alexander Kiselev's Juggler implementation `IRTAlgorithm`
+// Subject to the terms in the LICENSE file found in the top-level directory.
+
+#include "IrtCherenkovParticleID.h"
+
+// AlgorithmInit
+//---------------------------------------------------------------------------
+void eicrecon::IrtCherenkovParticleID::AlgorithmInit(
+    CherenkovDetectorCollection*     irt_det_coll,
+    std::shared_ptr<spdlog::logger>& logger
+    )
+{
+  // members
+  m_irt_det_coll = irt_det_coll;
+  m_log          = logger;
+  m_init_failed  = false;
+
+  // print the configuration parameters
+  m_cfg.Print(m_log, spdlog::level::debug);
+
+  // inform the user if a cheat mode is enabled
+  m_cfg.PrintCheats(m_log);
+
+  // extract the the relevant `CherenkovDetector`, set to `m_irt_det`
+  auto& detectors = m_irt_det_coll->GetDetectors();
+  if(detectors.size() == 0) {
+    m_log->error("No CherenkovDetectors found in input collection `irt_det_coll`");
+    m_init_failed = true;
```

Comment:
This case clearly not tested, followed up by dereferencing `detectors.begin()`. Please avoid adding dead code.

Reply:
These PID algorithms are designed to be callable from outside of EICrecon too. @alexander-kiselev prefers to store the IRT geometry, a `CherenkovDetectorCollection` object, in a ROOT file, and we want to maintain compatibility with that workflow here as much as possible. Checking `detectors.size()` is a useful sanity check when reading a ROOT file that contains a `CherenkovDetectorCollection`. Since this check is in `AlgorithmInit`, there is no performance loss.

Here in EICrecon, `CherenkovDetectorCollection` objects are generated by the `richgeo` service, where we guarantee `detectors.size()==1`; we do not generate such ROOT files, although we [maintain a program to do this](https://github.com/eic/drich-dev/blob/269bf3ceb1616dcf78aa5449262d4fc45f1b41b6/src/create_irt_auxfile.cpp#L1) for anyone who wants this feature.

So this code is only dead from the point of view of EICrecon usage.

Reply:
If `detectors.size() == 0`, it will crash on `auto this_detector = *detectors.begin();`, value of `m_init_failed` will never be used.

Reply:
Fixed: [c29fcbe](https://github.com/eic/EICrecon/pull/707/commits/c29fcbe452cbe54702c05663ce67e30d61d3fc7b)